icu4c/source/common/ucasemap.cpp

```diff
@@ -437,6 +437,9 @@ constexpr uint8_t ACUTE_BYTE1 = u8"\u0301"[1];
  */
 int32_t maybeTitleDutchIJ(const uint8_t *src, UChar32 c, int32_t start, int32_t segmentLimit,
                           ByteSink &sink, uint32_t options, icu::Edits *edits, UErrorCode &errorCode) {
+    if (start >= segmentLimit) {
```

Comment:
This UTF-8 version is called with `(start, segmentLimit)` = `titleLimit, index` (different from UTF-16) and this is inside the condition `if (titleStart+1 < index && ...` (just like in the UTF-16 version).

So here the problem is that when `c == u'Í'` then titleLimit==titleStart+2 (while in UTF-16 it's just +1).

I think the call site should be changed so that https://github.com/unicode-org/icu/blob/main/icu4c/source/common/ucasemap.cpp#L597 has `if (titleLimit < index &&`

Then you need not change this function.

Reply:
Done.  That makes sense.